### PR TITLE
Track wallet address for profile item

### PR DESCRIPTION
**Pull request by Devorein**
- [x] remove them when a user no longer has that wallet
- [x] Keep track of wallet address used to gather profile items

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -1025,13 +1026,15 @@ model ProposalReviewer {
...
-  type     ProfileItemType
-  isPinned Boolean?
+  id         String          @id
+  userId     String          @db.Uuid
+  user       User            @relation(fields: [userId], references: [id], onDelete: Cascade)
+  isHidden   Boolean?
+  metadata   Json?
+  type       ProfileItemType
+  isPinned   Boolean?
+  address    String?
```
Comment (mattcasey) [inline prisma/schema.prisma:1036]: change this to "walletAddress"? 'address' might be confused with 'contractAddress' too

Comment (mattcasey) [inline prisma/schema.prisma:1036]: rename this to 'walletAddress'? 'address' could be confused with 'contractAddress'

File: scripts/migrations/2023_03_30_add_default_path.ts
Comment (mattcasey) [inline scripts/migrations/2023_03_30_add_default_path.ts:1]: this migration should not be deleted

File: scripts/migrations/2023_04_05_add_profile_item_wallet.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+  })
+
+  for (const profileItem of profileItems) {
+    await prisma.profileItem.update({
+      where: {
+        id: profileItem.id
+      },
+      data: {
+        address: profileItem.user.wallets[0].address,
```
Comment (mattcasey) [inline scripts/migrations/2023_04_05_add_profile_item_wallet.ts:25]: we should check if a user has more than one wallet. i know of at least one user besides myself with multiple wallets. if they have more than one, i think it's safe to just skip them and not tie the profile items to a wallet

File: pages/api/profile/items.ts [hunk 3]
```diff
@@ -13,9 +13,23 @@ handler.use(requireUser).put(updateUserProfileItems);
...
+    select: {
+      wallets: {
+        select: {
+          address: true
+        }
+      }
+    }
+  });
+
+  const firstWalletAddress = userWallets?.wallets[0].address;
```
Comment (mattcasey) [inline pages/api/profile/items.ts:29]: the wallet address needs to be passed in, let's not assume which address it is. that defeats the purpose of tracking wallet address :P

File: pages/api/profile/items.ts [hunk 4]
```diff
@@ -54,7 +68,8 @@ async function updateUserProfileItems(req: NextApiRequest, res: NextApiResponse<
             userId: req.session.user.id,
             metadata: profileItem.metadata === null ? undefined : profileItem.metadata,
             isHidden: true,
-            type: profileItem.type
+            type: profileItem.type,
+            address: firstWalletAddress
```
Comment (mattcasey) [inline pages/api/profile/items.ts:72]: I guess this should actually just be profileItem.address

**Review by mattcasey — CHANGES_REQUESTED** [review]
It looks like we send profileItems to the frontend for users that don't own the profile items. To be safe, we should add a new unique key to wallets that we can link the profile items to, instead of using wallet addresss as the foreign key

File: scripts/migrations/2023_03_30_add_default_path.ts [hunk 5]
```diff
@@ -1,26 +0,0 @@
-import { prisma } from 'db';
-import { uid } from 'lib/utilities/strings';
-
-async function init() {
-  // const users = await prisma.user.findMany({
```
[the remark this answers is not in the document]
Comment (Devorein) [reply scripts/migrations/2023_03_30_add_default_path.ts:5]: As far as I remember ts was complaining, but I will remove the comments and see if it still does the same or not.

File: lib/users/prepopulateUserProfile.ts [hunk 6]
```diff
@@ -18,7 +18,10 @@ const acceptedImageFormats = ['.jpg', '.jpeg', '.png', '.webp'];
  * @param user we need the user id and user avatar to populate many
  * @param ens the ENS name
  */
-export async function prepopulateUserProfile(user: User, ens: string | null) {
+export async function prepopulateUserProfile(
+  user: User & { wallets: Pick<UserWallet, 'address' | 'ensname' | 'id'>[] },
```
Comment (mattcasey) [inline lib/users/prepopulateUserProfile.ts:22]: it looks like we only need the 'id' of wallet

File: pages/api/profile/items.ts [hunk 7]
```diff
@@ -14,6 +15,24 @@ handler.use(requireUser).put(updateUserProfileItems);
...
+      wallets: {
+        select: {
+          id: true
+        }
+      }
+    }
+  });
+
+  const userWalletIds = userWallets.wallets.map((wallet) => wallet.id);
+  if (profileItems.some((profileItem) => !profileItem.walletId || !userWalletIds.includes(profileItem.walletId))) {
```
Comment (mattcasey) [inline pages/api/profile/items.ts:32]: i'm not sure this logic is correct... shouldn't it be "profileItem.walletId && !userWallets.includes(profileItem.walletId)"?

**Review by mattcasey — APPROVED** [review]

**Pull request merged**